superset/cli.py

```diff
@@ -351,12 +351,17 @@ def update_datasources_cache():
     """Refresh sqllab datasources cache"""
     from superset.models.core import Database
     for database in db.session.query(Database).all():
-        print('Fetching {} datasources ...'.format(database.name))
-        try:
-            database.all_table_names(force=True)
-            database.all_view_names(force=True)
-        except Exception as e:
-            print('{}'.format(str(e)))
+        if database.allow_multi_schema_metadata_fetch:
+            print('Fetching {} datasources ...'.format(database.name))
+            try:
+                database.all_table_names_in_database(
+                    db_id=database.id, force=True,
```

Comment:
Why do you need to pass `database.id` here, if this is a method of `database`? Looking at other places where this method is called, looks like `db_id` is always the same as `database.id`, no?